llvm/include/llvm/Option/OptStrCmp.h

```diff
@@ -0,0 +1,51 @@
+//===- OptStrCmp.h - Option String Comparison -------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_OPTION_OPTSTRCMP_H
+#define LLVM_OPTION_OPTSTRCMP_H
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace llvm::opt {
+
+// Comparison function for Option strings (option names & prefixes).
+// The ordering is *almost* case-insensitive lexicographic, with an exception.
+// '\0' comes at the end of the alphabet instead of the beginning (thus options
+// precede any other options which prefix them). Additionally, if two options
+// are identical ignoring case, they are ordered according to case sensitive
+// ordering if `FallbackCaseSensitive` is true.
+static int StrCmpOptionName(StringRef A, StringRef B,

```

Comment:
This should probably lose `static` and either get `inline` or the implementation should live in a separate C++ file, right?

Reply:
I made it inline.

Reply:
I don't see either of the functions being `inline` in the latest revision. I'd suggest just moving the implementations to a `.cpp` file though.

Reply:
Right, looks like I undid it by mistake :( I do agree though that moving to CPP is the right thing here and I did try it out yestdrday. I do have a question about that. If I just do the simple thing and move it to `LLVMOption` component, I'd have to link in stuff into TableGen that it does not need today (from LLVMOption). If I make it a separate component, I'd have to use PARTIAL_SOURCES_INTENDED, which the build says is not recommended. So I guess I'd move it to a new directory under Options (so llvm/include/Option/OptionStrCmp/OptionStrCmp.h and llvm/lib/Option/OptionStrCmp/OptionStrCmp.cpp). Does that sound ok, or we just link LLVMOption into llvm-tablegen directly?

Reply:
Support sounds good! I'll work on updating the patch.

Reply:
Thanks!